dataset/src/main/scala/frameless/TypedEncoder.scala

```diff
@@ -296,24 +337,141 @@ object TypedEncoder {
     }
   }
 
+  implicit def arrayEncoder[A : ClassTag](
+    implicit
+    underlying: TypedEncoder[A]
+  ): TypedEncoder[Array[A]] = new TypedEncoder[Array[A]]() {
+    val nullable: Boolean = false
+
+    val sourceDataType: DataType = FramelessInternals.objectTypeFor[Array[A]](classTag)
+
+    val targetDataType: DataType = DataTypes.createArrayType(underlying.targetDataType)
+
+    def constructorFor(path: Expression): Expression = {
+      Option(underlying.sourceDataType)
```

Comment:
Could we factor this out with https://github.com/typelevel/frameless/pull/113/commits/d929233efd7bafad4759aee569710c87fc968064#diff-f0794dc7ae1a26e5c09857691a6b8855R293?